test/the-rewarder/README.md

```diff
@@ -0,0 +1,587 @@
+```solidity
+merkle = new Merkle();
+console.logBytes32(merkle.getRoot(dvtLeaves));
+console.logBytes32(merkle.getRoot(wethLeaves));
+...
+```
+And this is what we got:
+```javascript
+$ forge test --mp test/the-rewarder/TheRewarder.t.sol -vvv
+...
+Logs:
+    0x399df90cbebfb0e630b6da99a45325404a758823effc616197f3c33f749cb5d4
+    0x5a1b4e345b2e4419e385fa460b91decd0d9d34cac0bd187aedea5484d2cdd6f6
+    ...
+```
+So, Halmos test:
+```solidity
+merkle = new Merkle();
+dvtRoot = hex"399df90cbebfb0e630b6da99a45325404a758823effc616197f3c33f749cb5d4";
+wethRoot = hex"5a1b4e345b2e4419e385fa460b91decd0d9d34cac0bd187aedea5484d2cdd6f6";
+//dvtRoot = merkle.getRoot(dvtLeaves);
+//wethRoot = merkle.getRoot(wethLeaves);
+```
+Let's do a similar trick with Alice's `merkle.getProof()`. Halmos test is changed to:
+```solidity
+// Create Alice's claims
+Claim[] memory claims = new Claim[](2);
+bytes32[] memory dvtproof = new bytes32[](3);
+dvtproof[0] = hex"925450a3cfe3826ad85358e2b3df638edc7c8553b6faee9e40fd9c6e9e3a3e04";
+dvtproof[1] = hex"f262e0db29c13826883ed5262d51ad286f1bd627b4632141534c6cb80f01f430";
+dvtproof[2] = hex"5ad8d27e776667615f79b7c7be79980ac8352518ca274a8ed68a9953ee4302d5";
+
+bytes32[] memory wethproof = new bytes32[](3);
+wethproof[0] = hex"7217ae40b137a0d9d7179ef8bb0d0a0a8002dc6fefed8e9faa17b29bc037b747";
+wethproof[1] = hex"fdad7418265f24fd2100fbcde33a22785f151aa01ab26aefd76c58bbfa0a9592";
+wethproof[2] = hex"0be25e66daab92e7052e6c307ae4743bba49ae08c7324acbc3eb730f51b991e0";
+
+// First, the DVT claim
+claims[0] = Claim({
+    batchNumber: 0, // claim corresponds to first DVT batch
+    amount: ALICE_DVT_CLAIM_AMOUNT,
+    tokenIndex: 0, // claim corresponds to first token in `tokensToClaim` array
+    proof: dvtproof // Alice's address is at index 2
+});
+console.log("claims[0] created");
+
+// And then, the WETH claim
+claims[1] = Claim({
+    batchNumber: 0, // claim corresponds to first WETH batch
+    amount: ALICE_WETH_CLAIM_AMOUNT,
+    tokenIndex: 1, // claim corresponds to second token in `tokensToClaim` array
+    proof: wethproof // Alice's address is at index 2
+});
+```
+And another such place is `MerkleProof.verify()` from `TheRewarderDistributor::claimRewards()`:
+```solidity
+function claimRewards(Claim[] memory inputClaims, IERC20[] memory inputTokens) external {
+    ...
+    bytes32 leaf = keccak256(abi.encodePacked(msg.sender, inputClaim.amount));
+    bytes32 root = distributions[token].roots[inputClaim.batchNumber];
+    if (!MerkleProof.verify(inputClaim.proof, root, leaf)) revert InvalidProof();
+    ...
+}
+```
+Processing cryptographic proofs is very difficult for Halmos. Let's just remove this check, but at the same time assume that we have entered the correct data. We did something similar in [Naive-receiver](https://github.com/igorganich/damn-vulnerable-defi-halmos/tree/master/test/naive-receiver#optimizations).
```

Comment:
it would be interesting to discuss what it means to remove this check. IIUC, it means that we assume that an attacker is able to produce a valid merkle proof, and so we are not looking for a bug in the merkle verification procedure. Does that sound right? (and reasonable?)

Reply:
Described